### PR TITLE
Test for EFA Install

### PR DESCRIPTION
**Pull request by sean-smith**
Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: recipes/tests.rb [hunk 1]
```diff
@@ -99,6 +99,13 @@
   end
 end
 
+if node['cfncluster']['os'] == 'alinux' || node['cfncluster']['os'] == 'centos7'
+  execute 'check efa_install' do
```
Comment (lukeseawalker) [inline recipes/tests.rb]: execute 'ls efa root' do

File: recipes/tests.rb [hunk 2]
```diff
@@ -99,6 +99,13 @@
...
+    command "ls /opt/amazon/efa || exit 1"
```
Comment (lukeseawalker) [inline recipes/tests.rb]: I think "exit 1" is redundant

File: recipes/tests.rb [hunk 3]
```diff
@@ -99,6 +99,13 @@
...
+    user node['cfncluster']['cfn_cluster_user']
+  end
+end
+
```
Comment (lukeseawalker) [inline recipes/tests.rb]: What about the check on the driver (the one instance type dependent)?

File: recipes/tests.rb (hunk 1 above)
Comment (sean-smith) [reply recipes/tests.rb]: done

File: recipes/tests.rb (hunk 2 above)
Comment (sean-smith) [reply recipes/tests.rb]: yep, `ls` doesn't exit with a non-zero exit code if the directory doesn't exist, so I've changed it to use `-d` to check if the directory exists.

File: recipes/tests.rb (hunk 3 above)
Comment (sean-smith) [reply recipes/tests.rb]: As we discussed last week, the driver needs to be enabled in the LaunchTemplate which we can't change for the kitchen tests. This'll be tested in the integration test.

File: recipes/tests.rb (hunk 2 above)
Comment (sean-smith) [reply recipes/tests.rb]: I've changed this to check for the rpm being installed. `rpm -qa` lists installed rpms. See [1]

```bash
$ rpm -qa | grep libfabric
libfabric-devel-1.7.0amzn1.1-1.amzn2.x86_64
libfabric-1.7.0amzn1.1-1.amzn2.x86_64
$ rpm -qa | grep efa-
efa-0.9.2-1.amzn2.x86_64
```

[1] https://www.cyberciti.biz/howto/question/linux/linux-rpm-cheat-sheet.php

**Pull request merged**